### ISSUE TITLE
Error parsing large AMBER trajectories

### ISSUE THREAD

**Issue by marianoastor** [ParmEd/ParmEd#1131]
I am using a script to process relatively large AMBER trajectories. In a simplified way, it reads:

```
import parmed as pmd
trajectory = pmd.load_file("traj.nc")
```

If the trajectory is short (< ~125 frames) it works without problems. However, for longer trajectories, I get:

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/tokenize.py", line 330, in find_cookie
    line_string = line.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc8 in position 7: invalid continuation byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mac/amber18/lib/python2.7/site-packages/ParmEd-3.2.0-py2.7-linux-x86_64.egg/parmed/formats/registry.py", line 176, in load_file
    if cls.id_format(filename):
  File "/home/mac/amber18/lib/python2.7/site-packages/ParmEd-3.2.0-py2.7-linux-x86_64.egg/parmed/formats/sdf.py", line 31, in id_format
    words = linecache.getline(filename, lineno=4).split()
  File "/usr/lib/python3.8/linecache.py", line 16, in getline
    lines = getlines(filename, module_globals)
  File "/usr/lib/python3.8/linecache.py", line 47, in getlines
    return updatecache(filename, module_globals)
  File "/usr/lib/python3.8/linecache.py", line 136, in updatecache
    with tokenize.open(fullname) as fp:
  File "/usr/lib/python3.8/tokenize.py", line 394, in open
    encoding, lines = detect_encoding(buffer.readline)
  File "/usr/lib/python3.8/tokenize.py", line 371, in detect_encoding
    encoding = find_cookie(first)
  File "/usr/lib/python3.8/tokenize.py", line 335, in find_cookie
    raise SyntaxError(msg)
SyntaxError: invalid or missing encoding declaration for 'traj.nc'
```

Seeing that the problem occurs in sdf.py, I was able to overcome the problem by changing there:

```
        def id_format(filename):
        """ Identify the file as a SDF file format or not

        Parameters
        ----------
        filename : str
            Name of the file to test whether or not it is a sdf file

        Returns
        -------
        is_fmt : bool
            True if it is a sdf file, False otherwise
        """
        words = linecache.getline(filename, lineno=4).split()
        return len(words) >= 3 and words[-1] in ('V2000', 'V3000')
```

to

```
        def id_format(filename):
        """ Identify the file as a SDF file format or not

        Parameters
        ----------
        filename : str
            Name of the file to test whether or not it is a sdf file

        Returns
        -------
        is_fmt : bool
            True if it is a sdf file, False otherwise
        """
        try:
            words = linecache.getline(filename, lineno=4).split()
            return len(words) >= 3 and words[-1] in ('V2000', 'V3000')
        except SyntaxError:
            return False
```

I hope this helps.

**Comment by swails** [comment]
Thanks!

**Comment by swails** [comment]
I tried a fix in #1132
